Automattic/jetpack
# Update error notices my jetpack to use a whitelist

## Proposed changes:

* Add error notice whitelist so we don't show an error notice for all queries
* I decided which queries to throw an error for by looking at the old redux code and seeing if a global notice was created on a failed attempt of each query

### Other information:


## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via your local dev environment
2. Make sure your site is user connected
3. In `projects/packages/my-jetpack/src/class-rest-purchases.php` add the following line to the `get_site_current_purchases` function to return an error early
```php
return new \WP_Error( 'site_data_fetch_failed', 'Site data fetch failed', array( 'status' => $response_code ? $response_code : 400 ) );
```
4. Make sure the purchases error shows up as it should
![image](https://github.com/Automattic/jetpack/assets/65001528/06cc8a70-4d4b-492c-b5cc-49747855789d)
5. Now remove that error line and head to `projects/packages/my-jetpack/src/class-rest-zendesk-chat.php` and return an error early in the `get_chat_authentication` function
```php
		return new \WP_Error( 'chat_authentication_failed', 'Chat authentication failed', array( 'status' => $response_code ) );
```
6. Make sure the API call fails and that no error shows
![image](https://github.com/Automattic/jetpack/assets/65001528/98e3a0c7-d94c-425a-b0df-be0e04ff7d6e)
